dll/win32/shell32/wine/control.c

```diff
@@ -736,6 +737,107 @@ static	void	Control_DoWindow(CPanel* panel, HWND hWnd, HINSTANCE hInst)
 }
 #endif
 
+#ifdef __REACTOS__
+
+/** Structure for in and out data when
+ *  search for the cpl dialog of first instance
+ */
+typedef struct tagAppDlgFindData
+{
+    PCWSTR  szAppFile;  /**< Full path to applet library as search parameter */
+    signed  sAppletNo;  /**< Number of applet in a system control library as search parameter */
+    ATOM    aCPLName;   /**< to read window property 'CPLName' */
+    ATOM    aCPLFlags;  /**< to read window property 'CPLFlags'*/
+    HWND    hRunDLL;    /**< to skip self instance */
+    HWND    hDlgResult; /**< Returned dialog handle or NULL if not found */
+} AppDlgFindData;
+
+/*
+ * Case insensitive variant of wcsstr
+ */
+static
+wchar_t * wcsistr(const wchar_t *s, const wchar_t *b)
+{
+    wchar_t *x;
+    wchar_t *y;
+    wchar_t *c;
+    x = (wchar_t *)s;
+    while (*x)
+    {
+        if (towlower(*x) == towlower(*b))
+        {
+            y = x;
+            c = (wchar_t *)b;
+            while (*y && *c && towlower(*y) == towlower(*c))
+            {
+                c++;
+                y++;
+            }
+            if (!*c)
+                return x;
+        }
+        x++;
+    }
+    return NULL;
+}
+
+/**
+ * Callback function to search applet dialog
+ * @param hwnd A handle to a top-level window.
+ * @param lParam Pointer of AppDlgFindData
+ * @return TRUE: continue enumeration, FALSE: stop enumeration
+ */
+static BOOL CALLBACK
+Control_EnumWinProc(
+    _In_ HWND   hwnd,
+    _In_ LPARAM lParam
+)
+{
+    AppDlgFindData* pData;
+    pData = (AppDlgFindData*)lParam;
+    
+    if(pData->hRunDLL == hwnd)
+    {
+        // Skip self instance
+        return TRUE;
+    }
+    WCHAR szClassName[256] = L"";
+    if (GetClassNameW(hwnd, szClassName, _countof(szClassName)))
+    {
+        // Note: A comparison on identical is not possible, the class names are different.
+        // ReactOS: 'rundll32_window'
+        // WinXP: 'RunDLL'
+        // other OS: not checked
+        if (wcsistr(szClassName, L"rundll32") != NULL)
+        {
+            signed sAppletNo;
+
+            sAppletNo = (signed)GetPropW(hwnd, (LPTSTR)MAKELONG(pData->aCPLFlags, 0));
+            if (sAppletNo == pData->sAppletNo)
+            {
+                HANDLE hRes;
+                WCHAR szAppFile[MAX_PATH];
+
+                hRes = GetPropW(hwnd, (LPTSTR)MAKELONG(pData->aCPLName, 0));
+                GlobalGetAtomNameW((ATOM)(LONG)hRes, szAppFile, MAX_PATH);
+                if (wcscmp(szAppFile, pData->szAppFile) == 0)
+                {
+                    HWND hDialog;
+
+                    hDialog = GetLastActivePopup(hwnd);
+                    if (IsWindow(hDialog))
+                    {
+                        pData->hDlgResult = hDialog;
+                        return FALSE;// stop enumeration
+                    }
+                }
+            }
+        }
+    }
+    return TRUE; // continue enumeration
+}
+#endif
```

Comment:
(Putting what the closing endif refers to; see what is usually done in our codebase.)
```suggestion
#endif /* __REACTOS__ */
```